plenum/server/batch_handlers/audit_batch_handler.py

```diff
@@ -108,13 +108,12 @@ def _create_audit_txn_data(self, three_pc_batch, last_audit_txn):
 
         return txn
 
-    def __fill_ledger_root_hash(self, txn, three_pc_batch, lid, last_audit_txn):
-        target_ledger_id = three_pc_batch.ledger_id
+    def __fill_ledger_root_hash(self, txn, lid, ledger, last_audit_txn):
         last_audit_txn_data = get_payload_data(last_audit_txn) if last_audit_txn is not None else None
 
         # 1. ledger is changed in this batch => root_hash
-        if lid == target_ledger_id:
-            txn[AUDIT_TXN_LEDGER_ROOT][lid] = Ledger.hashToStr(three_pc_batch.txn_root)
+        if ledger.uncommittedTxns:
```

Comment:
We may have multiple uncommitted txns for the same ledger in fly, so that a check `if ledger.uncommittedTxns` doesn't man that a ledger was changed in this batch.

Reply:
Fixed